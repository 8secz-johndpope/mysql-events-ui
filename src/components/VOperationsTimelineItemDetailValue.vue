<script>
export default {
  name: 'OperationsTimelineItemDetailValue',
  props: {
    prefix: {
      type: String,
      default: '',
    },
    name: {
      type: String,
      required: true,
    },
    value: {
      required: true,
    },
    highlight: {
      type: Boolean,
      default: false,
    },
    add: {
      type: Boolean,
      default: false,
    },
    remove: {
      type: Boolean,
      default: false,
    }
  }
}
</script>


<template>
  <div
    :class="{
      'value--add': add,
      'value--remove': remove,
    }"
    class="value"
  >
    {{ prefix }}
    <span :class="{
      'highlight': highlight
    }">
      <strong>{{ name }}:</strong>
      {{ value }}
    </span>
  </div>
</template>

<style scoped>
.value {
  padding-left: 10px;
  padding-top: 5px;
  padding-bottom: 5px;
  text-align: left;
}

.value--remove {
  background-color: #ffeef0;
}

.value--add {
  background-color: #e6ffed;
}

.highlight {
  background-color: #acf2bd;
}
</style>
