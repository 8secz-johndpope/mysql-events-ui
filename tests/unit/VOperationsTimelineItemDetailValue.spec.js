import { shallowMount } from '@vue/test-utils'
import VOperationsTimelineItemDetailValue from '@/components/VOperationsTimelineItemDetailValue'
import socketFixture from './fixtures/socketServer'

describe('VOperationsTimelineItem', () => {
  let props

  const build = () => {
    const wrapper = shallowMount(VOperationsTimelineItemDetailValue, {
      propsData: props,
    })

    return {
      wrapper,
    }
  }

  beforeEach(() => {
    props = {
      name: 'id',
      value: 4,
    }
  })

  it('renders the component correctly', () => {
    // arranje
    const { wrapper } = build()

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})