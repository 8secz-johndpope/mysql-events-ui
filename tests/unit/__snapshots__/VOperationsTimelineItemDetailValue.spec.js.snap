// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VOperationsTimelineItem renders the component correctly 1`] = `
<div class="value">

  <span class=""><strong>id:</strong>
    4
  </span></div>
`;
